src/flask/app.py

```diff
@@ -322,7 +323,7 @@ def send_static_file(self, filename: str) -> Response:
 
     def open_resource(
         self, resource: str, mode: str = "rb", encoding: str | None = None
-    ) -> t.IO[t.AnyStr]:
+    ) -> t.IO[t.AnyStr] | BufferedReader:
```

Comment:
It seems like this should be reported to Pyright. Why doesn't `t.IO` satisfy buffered reader? Why is this a union, if `t.IO` isn't correct; shouldn't it be only `BufferedReader`?

Reply:
OK, so you're typeshed says that we're returning `BufferedReader`. But is that a guarantee we want to make vs returning the more general `IO`? I need more information to understand this change, "it passes pyright" isn't enough because it also changes what we report our API to be.